services/user/user.proto

```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
+
+package user;
+
+service UserService {
+  rpc Login(LoginRequest) returns (LoginResponse);
+}
+
+message LoginRequest {
+  string username = 1;
+  string hashed_password = 2;
```

Comment:
I believe the hashing process should be done on the login micro-service, no?

Reply:
You're correct the micro-service MUST hash. But it doesn't stop the front-end from sending an already hashed PW so that any leak in the way also does not give a plaintext password.